client/components/LearnerViewDirective.js

```diff
@@ -568,6 +586,33 @@ tie.directive('learnerView', [function() {
           }
         };
 
+        $scope.changeQuestionSet = function() {
+          questionSetId = '';
+          switch ($scope.currentQuestionSetId) {
+          case 'strings':
+          case 'other':
+          case 'all':
+            questionSetId = $scope.currentQuestionSetId;
+            break;
+          default:
+            return;
+          }
+          QuestionDataService.initCurrentQuestionSet(questionSetId);
+          questionSet = QuestionDataService.getCurrentQuestionSet(
+            questionSetId);
+          $scope.currentQuestionIndex = 0;
```

Comment:
Don't copy/paste. Extract all this setup stuff into another method (initQuestionSet, perhaps).